provider/provider_test.go

```diff
@@ -460,6 +460,13 @@ func configureProviderAndReturnClient(t *testing.T, tt providerFixture) (*common
 		return nil, fmt.Errorf(strings.Join(issues, ", "))
 	}
 	client := p.Meta().(*common.DatabricksClient)
-
+	r, err := http.NewRequest("GET", "", nil)
+	if err != nil {
+		return nil, err
+	}
+	err = client.Config.Authenticate(r)
+	if err != nil {
+		return nil, err
+	}
```

Comment:
This was removed in https://github.com/databricks/terraform-provider-databricks/commit/8842527bba1942d19cad5bb3bca57edb2bf9e416#diff-978ff4fdc176f5a02cba8681ff7f0c943bf0740a2ca7a68ff4e9f08d1efd3744L406-L410, but we add it back here to ensure existing test cases work. Those test cases make sure the provider auth works as expected when auth ought to work, but now we tolerate auth failing during provider construction, so we need to retry again to get the underlying auth error.